flink-runtime/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopFileSystem.java

```diff
@@ -459,17 +459,14 @@ public boolean isDistributedFS() {
 	@Override
 	public Class<?> getHadoopWrapperClassNameForFileSystem(String scheme) {
 		Configuration hadoopConf = getHadoopConfiguration();
-		Class<? extends org.apache.hadoop.fs.FileSystem> clazz;
-		// We can activate this block once we drop Hadoop1 support (only hd2 has the getFileSystemClass-method)
-//		try {
-//			clazz = org.apache.hadoop.fs.FileSystem.getFileSystemClass(scheme, hadoopConf);
-//		} catch (IOException e) {
-//			LOG.info("Flink could not load the Hadoop File system implementation for scheme "+scheme);
-//			return null;
-//		}
-		clazz = hadoopConf.getClass("fs." + scheme + ".impl", null, org.apache.hadoop.fs.FileSystem.class);
-
-		if (clazz != null && LOG.isDebugEnabled()) {
+		Class<? extends org.apache.hadoop.fs.FileSystem> clazz = null;
+		try {
+			clazz = org.apache.hadoop.fs.FileSystem.getFileSystemClass(scheme, hadoopConf);
+		} catch (IOException e) {
+			LOG.info("Flink could not load the Hadoop File system implementation for scheme " + scheme);
+		}
+
+		if(LOG.isDebugEnabled()) {
```

Comment:
Since clazz is null here is unreachable. ```getFileSystemClass``` will check null condition. And if ```clazz``` is null, ```getFileSystemClass``` will give a ```throw new IOException("No FileSystem for scheme: " + scheme); ``` instead. Another question, you said this style ? like ```if( LOG.isDebugEnabled() )``` ?